### PR TITLE
upgrade reference of .NET framework, System.Net.Http and NLog

### PR DESCRIPTION
**Pull request by bin3377**
update nuget packages to use:
1. System.Net.Http in 4.0 stable
2. .NET framework 4.5 (which is required by System.Net.Http)
3. Nlog in  4.3.3 (which is suggested to have 4.5 dependency)

### CODE REVIEW

**Comment by bin3377** [comment]
PR for issue https://github.com/SumoLogic/sumologic-net-appenders/issues/6
@bgoleno @latkin @CampsiteHelper

File: SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>SumoLogic.Logging.Common</RootNamespace>
     <AssemblyName>SumoLogic.Logging.Common</AssemblyName>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
```
Comment (latkin) [inline SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj]: Let's not add .NET 4.5 dependency - good to keep it on 4.0

File: SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs [hunk 2]
```diff
@@ -346,10 +346,10 @@ protected override void FlushAsync(global::NLog.Common.AsyncContinuation asyncCo
         {
             try
             {
-                var flushBufferTask = this.flushBufferTask;
-                if (flushBufferTask != null)
+                var task = this.flushBufferTask;
+                if (task != null)
                 {
-                    flushBufferTask.FlushAndSend();
+                    task.FlushAndSend();
```
Comment (latkin) [inline SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs]: Why the edits in this file?

File: SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj (hunk 1 above)
Comment (bin3377) [inline SumoLogic.Logging.Common/SumoLogic.Logging.Common.csproj]: then we need keep on System.Net.Http 2.x since 4.0 requires .NET 4.5

**Comment by latkin** [comment]
`packages` dir should be gitignored so that we don't have a ton of binary junk in the repo.

We should keep target framework at 4.0 so we don't change required framework version.

File: SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs (hunk 2 above)
Comment (bin3377) [inline SumoLogic.Logging.NLog/BufferedSumoLogicTarget.cs]: it's a warning from code analysis

**Comment by bin3377** [comment]
agree for package folder removing

**Pull request merged**